test/e2e/tests/reload.go

```diff
@@ -0,0 +1,229 @@
+			if err != nil {
+				t.Fatal(err) // Fail the test and log the error
+			}
+			var numReloads = 5
+
+			for i := 0; i < numReloads; i++ {
+				// Step 2: Restart or reload the Envoy Gateway
+				err := restartEnvoyGateway(suite.Client, namespace)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				// Step 3: Retrieve the `/config_dump` output from the Envoy Proxy admin interface
+				newConfigDump, err := getConfigDump(suite.RestConfig, suite.Client, namespace)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				// Step 4: Compare the obtained `/config_dump` output with the initial configuration
+				assert.Equal(t, initialConfig, newConfigDump, "Configuration mismatch after reload")
+
+				// Step 5: Repeat the above steps for the desired number of reloads
+			}
+		})
+	},
+}
+
+func getConfigDump(config *rest.Config, c client.Client, namespace string) (responseMap map[string]interface{}, err error) {
+	selectorLabels := map[string]string{
+		"gateway.envoyproxy.io/owning-gateway-name":      "all-namespaces",
+		"gateway.envoyproxy.io/owning-gateway-namespace": "gateway-conformance-infra",
+	}
+
+	// Create a new PodList to store the matching pods
+	podList := &corev1.PodList{}
+
+	// Build the ListOptions with the namespace and selectors
+	labelSelector := labels.SelectorFromSet(labels.Set(selectorLabels))
+	listOptions := &client.ListOptions{
+		Namespace:     namespace,
+		LabelSelector: labelSelector,
+	}
+
+	// List the pods using the ListOptions
+	err = c.List(context.TODO(), podList, listOptions)
+	if err != nil {
+		log.Fatal(err)
+		return nil, err
+	}
+
+	podName := podList.Items[0].Name
+
+	localPort := 19002
+	remotePort := 19000
+
+	transport, upgrader, err := spdy.RoundTripperFor(config)
+	if err != nil {
+		log.Fatal(err)
+		return nil, err
+	}
+
+	portForwardURL := fmt.Sprintf("/api/v1/namespaces/%s/pods/%s/portforward", namespace, podName)
+	portForwardURL = fmt.Sprintf("https://127.0.0.1:6443%s", portForwardURL)
+
+	serverURL, _ := url.Parse(portForwardURL)
+	ports := []string{fmt.Sprintf("%d:%d", localPort, remotePort)}
+
+	dialer := spdy.NewDialer(upgrader, &http.Client{Transport: transport}, http.MethodPost, serverURL)
+	stopCh := make(chan struct{}, 1)
+	readyCh := make(chan struct{})
+
+	// Create a port forwarder
+	portForwarder, err := portforward.New(dialer, ports, stopCh, readyCh, os.Stdout, os.Stderr)
+	if err != nil {
+		log.Fatal(err)
+		return nil, err
+	}
+
+	// Start port forwarding
+	go func() {
+		err := portForwarder.ForwardPorts()
+		if err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	// Wait until port forwarding is ready
+	<-readyCh
+
+	// Output the local address for accessing the forwarded port
+	fmt.Printf("Port forwarding started. Access the service locally at: localhost:%d\n", localPort)
+
+	// Perform an HTTP GET request to the forwarded port
+	resp, err := http.Get(fmt.Sprintf("http://localhost:%d/config_dump", localPort))
```

Comment:
can we also include eds ?
https://www.envoyproxy.io/docs/envoy/latest/operations/admin#get--config_dump?include_eds